From bf9dd5f3ee9e06a0e530ad442bd41e29be529c27 Mon Sep 17 00:00:00 2001
From: Yaroslav Halchenko <debian@onerussian.com>
Date: Fri, 23 Jul 2021 15:12:45 -0400
Subject: [PATCH] BF: import importlib.metadata not importlib_metadata whenever
 available

Bug introduced by myself in https://github.com/datalad/datalad/pull/5669 ,
which forbids building debian package on systems with python3 >= 3.8
where we do not (as should not) require importlib_metadata backport.
---
 _datalad_build_support/setup.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/_datalad_build_support/setup.py b/_datalad_build_support/setup.py
index 65a266fad..58dc7d811 100644
--- a/_datalad_build_support/setup.py
+++ b/_datalad_build_support/setup.py
@@ -40,7 +40,11 @@ def get_version(name):
       Name of the folder (package) where from to read version.py
     """
     # delay import so we do not require it for a simple setup stage
-    from importlib_metadata import version as importlib_version
+    try:
+        from importlib.metadata import version as importlib_version
+    except ImportError:
+        # TODO - remove whenever python >= 3.8
+        from importlib_metadata import version as importlib_version
     return importlib_version(name)
 
 
-- 
2.32.0

